secboot/secboot_tpm.go

```diff
@@ -438,7 +447,123 @@ func runFDERevealKeyCommand(stdin []byte) (output []byte, err error) {
 	return nil, fmt.Errorf("internal error: systemd-run did not honor RuntimeMax=%s setting", fdeRevealKeyRuntimeMax)
 }
 
+var noSecbootV2FileErr = errors.New("no secboot v2 file")
+
+// We have to deal with the following cases:
+// 1. Key created with v1 data-format on disk (raw encrypted key), v1 hook reads the data
+// 2. Key created with v2 data-format on disk (json), v1 hook created the data (no handle) and reads the data (hook output not json but raw binary data)
+// 3. Key created with v1 data-format on disk (raw), v2 hook
+// 4. Key created with v2 data-format on disk (json), v2 hook [easy]
 func unlockVolumeUsingSealedKeyFDERevealKey(name, sealedEncryptionKeyFile, sourceDevice, targetDevice, mapperName string, opts *UnlockVolumeUsingSealedKeyOptions) (UnlockResult, error) {
+
+	// deal with v1 keys
+	if isV1EncryptedKeyFile(sealedEncryptionKeyFile) {
+		return unlockVolumeUsingSealedKeyFDERevealKeyV1(name, sealedEncryptionKeyFile, sourceDevice, targetDevice, mapperName, opts)
+	}
+
+	return unlockVolumeUsingSealedKeyFDERevealKeyV2(name, sealedEncryptionKeyFile, sourceDevice, targetDevice, mapperName, opts)
+}
+
+// XXX: move to fde but right there there is a cirular import
```

Comment:
the circular import is nor more, so we have options about this now. I need to see how the split with kernel/fde will work out in practice though.